eclipse-theia/theia
# Update jsdiff and simplify diff computation

#### What it does

Fixes #13772

contributed on behalf of STMicroelectronics

#### How to test
* Open `electron.d.ts` as described in the issue and verify the UI is not blocked for a significant time.
* Edit a file and make sure the edits are represented properly in the "dirty diff annotation" you can invoke from the left editor gutter. 

#### Follow-ups
I think the performance is currently good enough on relevant scenarios that we don't have to move the diff computation into a worker, but if we do, that would be a followup.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
